bin/main.js

```diff
@@ -0,0 +1,270 @@
+          arg = argv.shift();
+        } else {
+          // e.g. -a
+        }
+      } else {
+        // e.g. foo
+      }
+
+      return arg;
+    }
+
+    while (argv.length) {
+      arg = getArg();
+      switch (arg) {
+        case '-o':
+        case '--output':
+          output = argv.shift();
+          break;
+        case '-i':
+        case '--input':
+          input = argv.shift();
+          break;
+        case '-s':
+        case '--string':
+          string = argv.shift();
+          break;
+        case '-t':
+        case '--tokens':
+          tokens = true;
+          break;
+        case '-c':
+        case '--config':
+          config = argv.shift();
+          break;
+        case '-h':
+        case '--help':
+          return await help();
+        case '-v':
+        case '--version':
+          return await version();
+        default:
+          if (arg.indexOf('--') === 0) {
+            opt = camelize(arg.replace(/^--(no-)?/, ''));
+            if (!marked.defaults.hasOwnProperty(opt)) {
+              continue;
+            }
+            if (arg.indexOf('--no-') === 0) {
+              options[opt] = typeof marked.defaults[opt] !== 'boolean'
+                ? null
+                : false;
+            } else {
+              options[opt] = typeof marked.defaults[opt] !== 'boolean'
+                ? argv.shift()
+                : true;
+            }
+          } else {
+            files.push(arg);
+          }
+          break;
+      }
+    }
+
+    async function getData() {
+      if (!input) {
+        if (files.length <= 2) {
+          if (string) {
+            return string;
+          }
+          return await getStdin();
+        }
+        input = files.pop();
+      }
+      return await readFile(input, 'utf8');
+    }
+
+    function resolveFile(file) {
+      return resolve(file.replace(/^~/, homedir));
+    }
+
+    function fileExists(file) {
+      return access(resolveFile(file)).then(() => true, () => false);
+    }
+
+    async function runConfig(file) {
+      const configFile = resolveFile(file);
+      let markedConfig;
+      try {
+        // try require for json
+        markedConfig = require(configFile);
+      } catch (err) {
+        if (err.code !== 'ERR_REQUIRE_ESM') {
+          throw err;
+        }
+        // must import esm
+        markedConfig = await import('file:///' + configFile);
+      }
+
+      if (markedConfig.default) {
+        markedConfig = markedConfig.default;
+      }
+
+      if (typeof markedConfig === 'function') {
+        markedConfig(marked);
+      } else {
+        marked.use(markedConfig);
+      }
+    }
+
+    const data = await getData();
+
+    if (config) {
+      if (!await fileExists(config)) {
+        throw Error(`Cannot load config file '${config}'`);
+      }
+
+      await runConfig(config);
+    } else {
+      const defaultConfig = [
+        '~/.marked.json',
+        '~/.marked.js',
+        '~/.marked/index.js'
+      ];
```

Comment:
Do we need to consider `.mjs` and `.cjs` here too?

It might be better to ship this with only `.json` first and see if JS support is actually needed.

Reply:
I think the `.js` support will be more useful to be able to include extensions. `.mjs` and `.cjs` works for the config option. I don't think we need to include them by default.